### ISSUE TITLE
 Raspberry Pi 4B with RTMP

### ISSUE THREAD

**Issue by Pellmart1** [Screenly/Anthias#1732]
#### Overview of the Issue

When adding a RTMP as an asset, it will not play.

#### Reproduction Steps

-access device using browser and IP
-click Add Asset
-enter url  rtmp://x.x.x.x:(port)/location
-add check to "Skip asset check" check box.
-asset gets added to list of assets
Get confirmation pop up : Asset has been successfully uploaded.
-New added assets in Inactive Assets list.  it will show the link but disabled.
-Change to enable
-Will show up in Active Assets list.
-Screen will flash, but nothing shows up but a random flashing white screen every minute or so.

If asset is added but no check is added to the Skip asset check box, then you get error pop up
"Server Error: global name 'mplayer' is not defined" 
It will not add the RTMP feed.
nothing changes on the screen.  simpl stays blank and accessible IP disapears.

#### Environment

Using Ethernet
Used Raspberry Pi Imager where you can find Anthias 

Option | Value
Load Average | 0.17
Free space | 22G
Memory | Total: 1731 \| Used: 412 \| Free: 801 \| Shared: 23 \| Buff: 38 \| Available: 1227
Uptime | 0 days and 0.16 hours
Monitor info | state 0xa [HDMI CUSTOM RGB lim 16:9], 1920x1080 @ 60.00Hz, progressive
Display Power (CEC) | None
Raspberry Pi Model | Raspberry Pi 4 Model B Rev 1.1
Screenly OSE Version | master@42b975e
MAC Address | Unable to retrieve MAC address.

Thanks for your time and consideration

**Comment by nicomiguelino** [comment]
Hello @Pellmart1. This feature (along with RSTP assets) has not been working and tested for a long time.

**Comment by vpetersson** [comment]
Yeah this is a known issue. Or rather, it was removed in the Pi 4 work. We plan to reinstate this in the future. But this won't happen until we're done with the Python 3/Django migration, which is the current priority.

**Comment by vpetersson** [comment]
Lol. You beat me to it @nicomiguelino.

**Comment by vpetersson** [comment]
Closing this out with an update.

Since the Python 3 / Django migration wrapped up, network streaming has been reinstated on the Pi 4 (and the other boards). The old `global name 'mplayer' is not defined` crash is gone too — the viewer now decodes through QtMultimedia/GStreamer rather than mplayer.

**RTSP streaming works today.** As of **v2026.07.0**, the viewer plays `rtsp://`, `srt://`, `udp://`, and `mms://` streams, plus HLS/DASH manifests (`.m3u8`, `.mpd`). Just add the stream URL as an asset — no "skip asset check" needed; the server validates and classifies it as a stream.

**RTMP specifically is intentionally not supported.** This isn't a codec limitation — FFmpeg decodes RTMP fine. The problem is that our viewer hands stream URLs to Qt6's `QMediaPlayer`, whose FFmpeg backend injects a socket `timeout` option on open. FFmpeg's `rtmp` protocol overloads `timeout` to mean "listen for an incoming connection," so instead of connecting to the origin the backend tries to listen and the open never completes — the screen just stays black. Rather than accept a URL that renders black, we reject `rtmp://` at asset-create time.

If you need RTMP, the practical workaround is a **local proxy**: run a small `ffmpeg` that pulls the `rtmp://` source and republishes it as RTSP or HLS locally, then point Anthias at that URL. FFmpeg has no trouble with RTMP ingest — it's only Qt's playback path that trips on it.

Given RTSP is the common path for cameras and live signage anyway, I'm closing this as not planned for native RTMP. If there's real demand for a built-in RTMP proxy, open a fresh feature request and we can look at it.
